src/app/pages/camper/schedule/schedule.component.ts

```diff
@@ -1,15 +1,75 @@
 import { Component, OnInit } from '@angular/core';
+import { ScheduleService } from "../../../service/schedule.service";
+import { interval } from 'rxjs';
 
 @Component({
   selector: 'app-schedule',
   templateUrl: './schedule.component.html',
   styleUrls: ['./schedule.component.scss']
 })
 export class ScheduleComponent implements OnInit {
+	
+  	private todayDate;
+  	private schedulelist;
+  	private hours;
+  	private mins;
+  	private secs;
 
-  constructor() { }
+  constructor(
+  	public schService:ScheduleService
+  	) { }
 
   ngOnInit() {
+  	this.schService.getSheet().subscribe(response => {
+
+	  	var date = ["20 JULY", "21 JULY", "22 JULY"];
+
+	  	var scheduleArr = response.values; 
+	  	this.schedulelist = this.getSchedule(scheduleArr);
+
+		interval(1000).subscribe(n =>
+			var eventDate = new Date(2018, 6, 20, 8, 0);
```

Comment:
a new Date object of the exact same parameters is constructed on every interval. 

this is akin to buying a new pen everyday when you go to school (which you probably don't need to cause you can buy 1 pen and use it for a long time) 

so instead of 
```javascript
interval(1000).subscribe(n => {
    let eventDate = new Date(2018, 6, 20, 8, 0);
    ...
});
```
You can do 
```javascript
// Since the date is never changing, const is even stricter which is good
const eventDate = new Date(2018, 6, 20, 8, 0);
interval(1000).subscribe(n => {
    ...
    const countdowntimer = eventDate.getTime() - currentTime.getTime();
});
```

This is about scoping which I can explain more in depth for you next time.

```